### PR TITLE
Project template language and messaging #1190, #1118

### PR DESCRIPTION
**Pull request by JimMarlowe**
Made JavaScript the default language used for new projects and examples ( if they support it )
Examples will use the user selected default language, and otherwise the language precedence is JS, CS, TS.
Upon successfully installing a new project or example, the language used will become the default language.
The default language is saved in a preference.
When CSharp is not installed on a platform, the dialog box that comes up will now include instructions for Linux+monodevelop.
The Linux install operation will be directed to the Wiki page for building linux, which includes the install instructions for monodevelop.
Added API to get item strings out of a UISelectItemSource
Did not fix the closing of the New Project Dialog if it brings up the CSharp Download dialog, because both are modal, and you can not have 2 modal dialogs up at a time.

### CODE REVIEW

File: Script/AtomicEditor/ui/modal/CreateProject.ts [hunk 1]
```diff
@@ -327,7 +329,12 @@ class CreateProject extends ModalWindow {
             }
             else if (id == "create") {
 
-                this.tryProjectCreate();
+                if ( this.tryProjectCreate() )
+                {
+                    Preferences.getInstance().editorFeatures.defaultLanguage = this.projectLanguageField.text;
+                    Preferences.getInstance().write();
```
Comment (JoshEngebretson) [inline Script/AtomicEditor/ui/modal/CreateProject.ts]: Can we write only in the case of it the preference being changed?

File: Script/AtomicEditor/ui/modal/CreateProject.ts [hunk 2]
```diff
@@ -366,6 +373,26 @@ class CreateProject extends ModalWindow {
...
+
+        if ( this.defaultLang == "JavaScript" ) defrank = jsrank; // which is the default language
+        if ( this.defaultLang == "CSharp" ) defrank = csrank;
+        if ( this.defaultLang == "TypeScript" ) defrank = tsrank;
+
+        if ( defrank > -1 ) this.projectLanguageField.value = defrank;  // the default language is present
+        else if ( jsrank > -1 ) this.projectLanguageField.value = jsrank;  // js is present
+        else if ( csrank > -1 ) this.projectLanguageField.value = csrank;  // cs is present
+        else if ( tsrank > -1 ) this.projectLanguageField.value = tsrank;  // ts is present
     }
```
Comment (JoshEngebretson) [inline Script/AtomicEditor/ui/modal/CreateProject.ts]: Fancy footwork 👣

File: Script/AtomicEditor/ui/modal/info/AtomicNETWindow.ts [hunk 3]
```diff
@@ -64,27 +64,38 @@ class AtomicNETWindow extends ModalWindow {
...
-                "https://www.visualstudio.com/vs/community/" :
-                "https://www.xamarin.com/download/");
+                if ( Atomic.platform == "Windows") 
+                    Atomic.fileSystem.systemOpen( "https://www.visualstudio.com/vs/community/" );
+
+                if ( Atomic.platform == "MacOSX") 
+                    Atomic.fileSystem.systemOpen( "https://www.xamarin.com/download/");
+
+                if ( Atomic.platform == "Linux") 
+                    Atomic.fileSystem.systemOpen( "https://github.com/AtomicGameEngine/AtomicGameEngine/wiki/Detailed-instructions-for-building-on-Linux");
```
Comment (JoshEngebretson) [inline Script/AtomicEditor/ui/modal/info/AtomicNETWindow.ts]: Wiki Power!

File: Source/Atomic/UI/UISelectItem.cpp [hunk 4]
```diff
@@ -107,6 +107,17 @@ void UISelectItemSource::RemoveItemWithStr(const String& str)
...
+{
+	int nn = 0;
+    for (List<SharedPtr<UISelectItem> >::Iterator itr = items_.Begin(); itr != items_.End(); itr++)
+    {
+        if ( nn == index) return (*itr)->GetStr();
+		nn++;
+    }
+	return ( String::EMPTY );
+}
+
```
Comment (JoshEngebretson) [inline Source/Atomic/UI/UISelectItem.cpp]: Tabs?

File: Source/Atomic/UI/UISelectItem.h [hunk 5]
```diff
@@ -80,6 +80,8 @@ class UISelectItemSource : public Object
 
     void Clear() { items_.Clear(); }
 
+    const String& GetItemStr(int index);
+
```
Comment (JoshEngebretson) [inline Source/Atomic/UI/UISelectItem.h]: I've been trying to remember to add /// comment blocks above methods when adding them

File: Script/AtomicEditor/ui/modal/CreateProject.ts (hunk 1 above)
Comment (JimMarlowe) [inline Script/AtomicEditor/ui/modal/CreateProject.ts]: Yes, I'll do that.

File: Script/AtomicEditor/ui/modal/info/AtomicNETWindow.ts (hunk 3 above)
Comment (JimMarlowe) [inline Script/AtomicEditor/ui/modal/info/AtomicNETWindow.ts]: :camel: time.

File: Source/Atomic/UI/UISelectItem.h (hunk 5 above)
Comment (JimMarlowe) [inline Source/Atomic/UI/UISelectItem.h]: The /// comments are normally for doxygen and other doc processors, if you'd like that to be the standard practice, I'll make a note of that and ensure it gets done.

Comment (JoshEngebretson) [inline Source/Atomic/UI/UISelectItem.h]: Yes, any function /// comment will automatically be added to C# and TS intellisense by the binding system :)  It is an inherited Urho convention too: https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Source/Atomic/Graphics/Graphics.h#L121

**Comment by JoshEngebretson** [comment]
Awesome! Major usability upgrade, and 2x issues closed, thanks! 👍 

Closes #1190 
Closes #1118

**Pull request merged**